src/Components/Common/ArgumentExceptionExtensions.cs

```diff
@@ -0,0 +1,45 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace Aspire;
+
+internal static class ArgumentExceptionExtensions
+{
+    /// <summary>
+    /// Adds a check Throw ArgumentException if argument is null or empty.
+    /// And return argument
+    /// </summary>
+    /// <param name="argument"></param>
+    /// <param name="paramName"></param>
+    /// <returns>argument not null</returns>
+    public static string ThrowIfNullOrEmpty(
```

Comment:
style wise, not sure whether folks want all strings and string arrays in the codebase to have new methods or not. (i don't think I care, but in dotnet/runtime they would prefer explicit static methods)

@mitchdenny any opinion?

Reply:
this team does love extension methods...

Reply:
I like that too.